### ISSUE TITLE
Bug: [require-await] currently doesn't show any errors or warnings even when it should hit

### ISSUE THREAD

**Issue by Flixbox** [typescript-eslint/typescript-eslint#11093]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.8.2&fileType=.ts&code=IYZwngdgxgBAZgV2gFwJYHsI1JKAxJKACgEoYBvAKBmwHdhVkYIBTWmABQCd0BbVECyKkYAXgB8MAIwlKAX0qUc0AtFJA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEI0QR1gEtpEBaAQwHdzCAXdKAewDNnIAacbSAAVoE8ADigDG0QoNqkU8QgDtaAehIFiZKjXoZIiaNEbRIXAL4hjQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
async function asyncFunc() {
  await new Promise(() => 1)
}

asyncFunc()
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "require-await": "off",
    "@typescript-eslint/require-await": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I would expect that the rule flags the issue (missing await) in line 5.

### Actual Result

There are no issues flagged.

### Additional Info

This could just be a missing configuration in the sandbox environment. I'm not quite sure. Currently trying to reproduce this locally, but ESLint just crashes when trying to report this rule (likely an unrelated local config issue, will create a separate issue if I get that specific problem reproduced in the sandbox)

**Comment by bradzacher** [comment]
> I would expect that the rule flags the issue.

WHAT issue?
Your body does not describe what you're expecting -- it just includes a code fragment?
What line would you expect to flag and **_WHY_**?

Your playground also has more code than your issue body lists -- which further confuses what line you're talking about.

Please -- help me to help you.

**Comment by Flixbox** [comment]
[Minimal playground link](https://typescript-eslint.io/play/#ts=5.8.2&fileType=.ts&code=IYZwngdgxgBAZgV2gFwJYHsI1JKAxJKACgEoYBvAKBmwHdhVkYIBTWmABQCd0BbVECyKkYAXgB8MAIwlKAX0qUc0AtFJA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEI0QR1gEtpEBaAQwHdzCAXdKAewDNnIAacbSAAVoE8ADigDG0QoNqkU8QgDtaAehIFiZKjXoZIiaNEbRIXAL4hjQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

Issue: The rule should flag the missing `await`.

I'm expecting this to be flagged when the function is called.

**Comment by bradzacher** [comment]
Have you had a look a the rule docs...?
https://typescript-eslint.io/rules/require-await/

> This rule warns async functions which have no await expression

So as described in my comment above -- the declaration of `asyncFunction` is correctly not reported on.

---

Are you instead expecting that the _call_ of `asyncFunction` _must_ be `await`ed?
If so -- that's a different rule.
https://typescript-eslint.io/rules/no-floating-promises
